lib/dependencies/RequireContextPlugin.js

```diff
@@ -30,15 +30,17 @@ class RequireContextPlugin {
 
 			compilation.dependencyFactories.set(ContextElementDependency, normalModuleFactory);
 
-			normalModuleFactory.plugin(["parser javascript/auto", "parser javascript/dynamic"], (parser, parserOptions) => {
-
+			const handler = (parser, parserOptions) => {
 				if(typeof parserOptions.requireContext !== "undefined" && !parserOptions.requireContext)
 					return;
 
 				parser.apply(new RequireContextDependencyParserPlugin());
-			});
+			};
+
+			normalModuleFactory.hooks.parser.for("javascript/auto").tap("RequireContextPlugin", handler);
+			normalModuleFactory.hooks.parser.for("javascript/dynamic").tap("RequireContextPlugin", handler);
 
-			contextModuleFactory.plugin("alternatives", (items, callback) => {
+			contextModuleFactory.hooks.alternatives.tapAsync("RequireContextPlugin", (items, callback) => {
```

Comment:
Use `tap` instead of `tapAsync` and remove the `callback`. Return the result instead.